freqtrade/freqtradebot.py

```diff
@@ -204,6 +200,24 @@ def _process(self) -> bool:
             self.state = State.STOPPED
         return state_changed
 
+    def _query_trades(self) -> List[Any]:
```

Comment:
nice cleanup!
I personally would move this to `persistence.py` and call it `Trade.get_open()` (or similar)...

also - `Trade.query.filter(Trade.is_open.is_(True)).all()` appears 3 times in this file alone, and another 3 times in `rpc.py` - if we move it to a function, we should replace all of them ...

Reply:
Actually I guess the whole Freqtradebot class should be split into 2 classes in the future in order to split the app start/stop/loop/reconfig logic and trading business logic which starts in the _process...

Reply:
> nice cleanup!
> I personally would move this to `persistence.py` and call it `Trade.get_open()` (or similar)...
> 
> also - `Trade.query.filter(Trade.is_open.is_(True)).all()` appears 3 times in this file alone, and another 3 times in `rpc.py` - if we move it to a function in persistence, we could replace all of them at once...

Fine... :+1:

Reply:
Absolutely, even more pieces if you will ... has been discussed in #1335 if you're looking for more about that